### PR TITLE
[fix]: [OIP-657]: Fixed error messages not propagating properly.

### PR DESCRIPTION
**Pull request by n00bitax**
## Describe your changes

- We were not propagating the actual error message and just giving the "Error" text in the response.
- We are relying on the exception error message to be present, which is incorrect.

Screenshot 2023-07-15 at 11 41 05 AM
Screenshot 2023-07-15 at 12 58 58 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50589)

### CODE REVIEW

**Comment by n00bitax** [comment]
trigger uts_JDK11

**Comment by n00bitax** [comment]
trigger uts_JDK11

**Review by naveen-harness — APPROVED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/cvng/CVNGDataCollectionDelegateServiceImpl.java [hunk 1]
```diff
@@ -146,6 +89,72 @@ public String getDataCollectionResult(String accountId, DataCollectionRequest da
...
+             decryptableEntityIndex++) {
+          DecryptableEntity decryptableEntity = decryptableEntities.get(decryptableEntityIndex);
+          List<EncryptedDataDetail> encryptedDataDetail = encryptedDataDetails.get(decryptableEntityIndex);
+          secretDecryptionService.decrypt(decryptableEntity, encryptedDataDetail);
+        }
+      }
+    }
+  }
+
+  private Consumer<CallDetails> getCallDetailsConsumer(
```
[the remark this answers is not in the document]
Comment (n00bitax) [reply 930-delegate-tasks/src/main/java/io/harness/cvng/CVNGDataCollectionDelegateServiceImpl.java:117]: how about APICallLogsPopulationLogicConsumer ?, I will take up in the next PR

File: 930-delegate-tasks/src/main/java/io/harness/cvng/CVNGDataCollectionDelegateServiceImpl.java [hunk 2]
```diff
@@ -96,46 +76,9 @@ public String getDataCollectionResult(String accountId, DataCollectionRequest da
...
-                                               .build());
-            }
-            if (callDetails.getRequest().request().body() != null) {
-              cvngLogDTO.addCallDetailsBodyFieldToRequest(callDetails.getRequest().request());
-            }
-            setApiCallLogDTOWithResponse(callDetails, cvngLogDTO);
-            delegateLogService.save(accountId, cvngLogDTO);
-          }
-        }));
+        Consumer<CallDetails> callDetailsConsumer = getCallDetailsConsumer(accountId, dataCollectionRequest, now);
```
Comment (AbhijithHarness) [inline 930-delegate-tasks/src/main/java/io/harness/cvng/CVNGDataCollectionDelegateServiceImpl.java:79]: Please change the variable name too.

File: 930-delegate-tasks/src/main/java/io/harness/cvng/CVNGDataCollectionDelegateServiceImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (AbhijithHarness) [reply 930-delegate-tasks/src/main/java/io/harness/cvng/CVNGDataCollectionDelegateServiceImpl.java:117]: CallDetailsLogger should be fine, (Logic is true for all code and consumer is optional for me, as we get it from type)

**Review by AbhijithHarness — APPROVED** [review]

**Pull request merged**